libbzip2-rs-sys/src/bzlib.rs

```diff
@@ -1624,26 +1624,31 @@ pub(crate) unsafe fn BZ2_bzDecompressHelp(strm: &mut bz_stream) -> ReturnCode {
 
         match s.state {
             decompress::State::BZ_X_IDLE | decompress::State::BZ_X_OUTPUT => continue,
-            _ => match decompress(strm, s) {
-                ReturnCode::BZ_STREAM_END => {
-                    if s.verbosity >= 3 {
-                        #[cfg(feature = "std")]
-                        std::eprint!(
-                            "\n    combined CRCs: stored = {:#08x}, computed = {:#08x}",
-                            s.storedCombinedCRC,
-                            s.calculatedCombinedCRC,
-                        );
-                    }
-                    if s.calculatedCombinedCRC != s.storedCombinedCRC {
-                        return ReturnCode::BZ_DATA_ERROR;
+            _ => {
+                let Some(allocator) = (unsafe { Allocator::from_bz_stream(strm) }) else {
+                    return ReturnCode::BZ_PARAM_ERROR;
+                };
```

Comment:
I think this should happen at the top of the function, so that param errors are returned early and we don't do any work for an invalid input. (the work can be observed with debug prints)